// @hub2/common/dist/types/dto/pagination.dto.js
"use strict";
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.OffsetPaginationQuery = exports.CursorPaginatedResponse = exports.OffsetPaginatedResponse = void 0;
const class_validator_1 = require("class-validator");
const transformer_1 = require("../../transformer");
class OffsetPaginatedResponse {
    constructor(data, total, page, limit) {
        this.data = data;
        this.total = total;
        this.page = page;
        this.limit = limit;
    }
}
exports.OffsetPaginatedResponse = OffsetPaginatedResponse;
class CursorPaginatedResponse {
    constructor(data, nextCursor, previousCursor) {
        this.data = data;
        this.nextCursor = nextCursor;
        this.previousCursor = previousCursor;
    }
}
exports.CursorPaginatedResponse = CursorPaginatedResponse;
class OffsetPaginationQuery {
}
__decorate([
    (0, transformer_1.ToInt)(),
    (0, class_validator_1.IsInt)(),
    __metadata("design:type", Number)
], OffsetPaginationQuery.prototype, "page", void 0);
__decorate([
    (0, transformer_1.ToInt)(),
    (0, class_validator_1.IsInt)(),
    __metadata("design:type", Number)
], OffsetPaginationQuery.prototype, "limit", void 0);
exports.OffsetPaginationQuery = OffsetPaginationQuery;
