// @hub2/common/dist/validator/anti.validator.js
"use strict";
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.Anti = exports.AntiConstraint = void 0;
const class_validator_1 = require("class-validator");
let AntiConstraint = class AntiConstraint {
    defaultMessage(validationArguments) {
        const [relatedPropertyName] = validationArguments === null || validationArguments === void 0 ? void 0 : validationArguments.constraints;
        return `$property cannot be used with ${relatedPropertyName}`;
    }
    validate(value, validationArguments) {
        const [relatedPropertyName] = validationArguments === null || validationArguments === void 0 ? void 0 : validationArguments.constraints;
        const relatedValue = validationArguments === null || validationArguments === void 0 ? void 0 : validationArguments.object[relatedPropertyName];
        return !(!!relatedValue && !!value);
    }
};
AntiConstraint = __decorate([
    (0, class_validator_1.ValidatorConstraint)({ name: 'Anti', async: false })
], AntiConstraint);
exports.AntiConstraint = AntiConstraint;
function Anti(property, validationOptions) {
    return function (object, propertyName) {
        (0, class_validator_1.registerDecorator)({
            name: 'anti',
            target: object.constructor,
            propertyName,
            constraints: [property],
            options: validationOptions,
            validator: AntiConstraint,
        });
    };
}
exports.Anti = Anti;
