// @hub2/common/dist/types/enums/filter-operator.enum.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.FilterOperator = void 0;
var FilterOperator;
(function (FilterOperator) {
    FilterOperator["EQUALS"] = "$eq";
    FilterOperator["NOT_EQUALS"] = "$ne";
    FilterOperator["GREATER_THAN"] = "$gt";
    FilterOperator["GREATER_THAN_OR_EQUALS"] = "$gte";
    FilterOperator["LOWER_THAN"] = "lt";
    FilterOperator["LOWER_THAN_OR_EQUALS"] = "$lte";
    FilterOperator["STARTS_WITH"] = "$starts";
    FilterOperator["ENDS_WITH"] = "$ends";
    FilterOperator["CONTAINS"] = "$cont";
    FilterOperator["EXCLUDES"] = "$excl";
    FilterOperator["IN"] = "$in";
    FilterOperator["NOT_IN"] = "$notin";
    FilterOperator["IS_NULL"] = "$isnull";
    FilterOperator["NOT_NULL"] = "$notnull";
    FilterOperator["BETWEEN"] = "$between";
    FilterOperator["EQUALS_INSENSITIVE"] = "$eqL";
    FilterOperator["NOT_EQUALS_INSENSITIVE"] = "$neL";
    FilterOperator["STARTS_WITH_INSENSITIVE"] = "$startsL";
    FilterOperator["ENDS_WITH_INSENSITIVE"] = "$endsL";
    FilterOperator["CONTAINS_INSENSITIVE"] = "$contL";
    FilterOperator["IN_INSENSITIVE"] = "$inL";
    FilterOperator["NOT_IN_INSENSITIVE"] = "$notinL";
})(FilterOperator = exports.FilterOperator || (exports.FilterOperator = {}));
